x/emissions/keeper/msgserver/msg_server_worker_payload_test.go

```diff
@@ -220,27 +241,38 @@ func (s *MsgServerTestSuite) TestMsgInsertWorkerPayloadFailsWithoutSignature() {
 	// workerMsg = s.signMsgInsertWorkerPayload(workerMsg, workerPrivateKey)
 	// END MODIFICATION
 
-	_, err := msgServer.InsertWorkerPayload(ctx, &workerMsg)
+	// Add worker to topic whitelist
+	err := s.emissionsKeeper.AddToTopicWorkerWhitelist(ctx, workerMsg.WorkerDataBundle.TopicId, workerMsg.WorkerDataBundle.Worker)
+	require.NoError(err)
+
+	_, err = msgServer.InsertWorkerPayload(ctx, &workerMsg)
 	require.ErrorIs(err, sdkerrors.ErrInvalidRequest)
 }
 
 func (s *MsgServerTestSuite) TestMsgInsertWorkerPayloadFailsWithMismatchedTopicId() {
 	ctx, msgServer := s.ctx, s.msgServer
 	require := s.Require()
+	topicId := uint64(123)
 
 	workerPrivateKey := secp256k1.GenPrivKey()
 
 	workerMsg, _ := s.setUpMsgInsertWorkerPayload(workerPrivateKey)
 
+	// Add worker to topic whitelist
+	err := s.emissionsKeeper.AddToTopicWorkerWhitelist(ctx, topicId, workerMsg.WorkerDataBundle.Worker)
+	require.NoError(err)
+	err = s.emissionsKeeper.AddToTopicWorkerWhitelist(ctx, workerMsg.WorkerDataBundle.TopicId, workerMsg.WorkerDataBundle.Worker)
```

Comment:
Duplicated two lines too?
If the purpose is to ensure that topicId is the same as workerMsg.WorkerDataBundle.TopicId, that should merit its own assertion.